lib/time.py

```diff
@@ -23,3 +55,48 @@ def sleep(secs):
 
 def time():
   return float(Now().UnixNano()) / Second
+
+
+_strftime_directive_map = {
+    'Y': '2006', 'y': '06', 'b': 'Jan', 'B': 'January', 'm': '01', 'd': '02',
+    'a': 'Mon', 'A': 'Monday',
+    'H': '15', 'I': '03', 'M': '04', 'S': '05', 'L': '.000',
+    'p': 'PM',
+    'Z':  'MST', 'z': '-0700',
+    '%': '%',
+    'c': NotImplementedError, 'j': NotImplementedError,
+    'U': NotImplementedError, 'w': NotImplementedError,
+    'W': NotImplementedError, 'x': NotImplementedError,
+    'X': NotImplementedError, '0': NotImplementedError,
+    '1': NotImplementedError, '2': NotImplementedError,
+    '3': NotImplementedError, '4': NotImplementedError,
+    '5': NotImplementedError, '6': NotImplementedError,
+    '7': NotImplementedError, '8': NotImplementedError,
+    '9': NotImplementedError,
+}
+
+def strftime(format, tt=None):  # pylint: disable=missing-docstring,redefined-builtin
+  t = (Unix(int(mktime(tt)), 0) if tt else Now()).Local()
+  ret = []
+  prev, n = 0, format.find('%', 0, -1)
+  while n != -1:
+    ret.append(format[prev:n])
+    next_ch = format[n+1]
```

Comment:
Spaces around operators, even inside subscript (this is different than the way subscripts are formatted in Go).